test/e2e/util.go

```diff
@@ -265,6 +265,34 @@ func pokeEndpointHostname(clientContainer, protocol, targetHost string, targetPo
 	return hostName
 }
 
+// pokeEndpointHostnameWithTimeout leverages a container running the netexec command to send a "hostname" request to a target running
+// netexec on the given target host / protocol / port.
+// Returns the name of backend pod.
+func pokeEndpointHostnameWithTimeout(clientContainer, protocol, targetHost string, targetPort int32, timeout int) string {
```

Comment:
why do we suddenly need a new function with 4 seconds timeout for this test while other tests are behaving fine with the default `pokeEndpointHostname` function? Indirectly I guess I am asking is there an intermediary period for this test where we know the curl won't work and we want to limit a curl timeout to avoid a hang forever situation?

Reply:
Yeah that was the idea, to give it some "buffer" before starting the actual tests but neither is it very clear from the code nor do I think that it's working that well.